server/game/cards/01_SOR/events/MakeAnOpening.ts

```diff
@@ -0,0 +1,32 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { EventCard } from '../../../core/card/EventCard';
+import { CardType, WildcardCardType } from '../../../core/Constants';
+
+export default class MakeAnOpening extends EventCard {
+    protected override getImplementationId () {
+        return {
+            id: '3208391441',
+            internalName: 'make-an-opening',
+        };
+    }
+
+    public override setupCardAbilities () {
+        this.setEventAbility({
+            title: 'Give a unit -2/-2 for this phase. Heal 2 damage from your base',
+            targetResolvers: {
+                downgradeUnit: {
+                    cardTypeFilter: WildcardCardType.Unit,
+                    immediateEffect: AbilityHelper.immediateEffects.forThisPhaseCardEffect({
+                        effect: AbilityHelper.ongoingEffects.modifyStats({ power: -2, hp: -2 })
+                    })
+                },
+                heal: {
```

Comment:
So this case might get a little complicated. The first thing is that I think you may have misread the card, it specifically says "heal your base" not "heal a base."

That said, since there is only one target for the effect, we need to change this. We only want to use target resolvers if there are potentially multiple targets for the player to choose. For the heal case there are not, the target is known in advance. This code works but it is kind of going about it slightly the wrong way.

In that case normally we use the `immediateEffect` field instead of `targetResolver`. I recommend you first take a look at the comment I left on Smuggler's Aid since that will kind of show the difference.

This case is more complex than Smuggler's Aid for two reasons:

1. There are multiple effects
2. We have one effect that needs a target decided by the player (-2/-2), and one that does not (heal)

To get some initial context, you should the read the wiki about multiple effects: https://github.com/SWU-Karabast/forceteki/wiki/Advanced-Card-Behaviors#abilities-with-multiple-effects

So what we need to do is use the `simultaneous()` effect to create two effects that resolve together, and then use `selectCard` to let the player choose the target for the -2/-2. It will look kind of like this:

```typescript
  this.setEventAbility({
      title: 'Give a unit -2/-2 for this phase. Heal 2 damage from your base',
      immediateEffect: AbilityHelper.immediateEffects.simultaneous([
          AbilityHelper.immediateEffects.selectCard({
              innerSystem: AbilityHelper.immediateEffects.forThisPhaseCardEffect({
                  effect: AbilityHelper.ongoingEffects.modifyStats({ power: -2, hp: -2 }),
              }),
          }),
          AbilityHelper.immediateEffects.heal((context) => ({ amount: 2, target: context.source.controller.base }))
      ]),
  });
```

Unfortunately there are not many cards yet that use `selectCard` so we don't have great examples for it, you could look at [Headhunting](https://github.com/SWU-Karabast/forceteki/blob/944e671f65218a4220b75afb3ef32495627f3c2d/server/game/cards/02_SHD/events/Headhunting.ts#L27) as an example but that's a complex one.

The tl;dr is basically it functions just like `targetResolver`, and it's what we have to use in situations where we are mixing things that need the player to choose a target (the -2/-2) and things that don't (the heal effect).